community/kernel/src/main/java/org/neo4j/kernel/impl/store/counts/keys/IndexKey.java

```diff
@@ -79,6 +70,17 @@ public boolean equals( Object other )
         }
 
         IndexKey indexKey = (IndexKey) other;
-        return labelId == indexKey.labelId && propertyKeyId == indexKey.propertyKeyId && type == indexKey.type;
+        return indexKey.descriptor.equals( descriptor );
+    }
+
+    @Override
+    public int compareTo( CountsKey other )
+    {
+        if ( other instanceof IndexKey )
+        {
+            IndexKey that = (IndexKey) other;
+            return this.descriptor.descriptor().compareTo( that.descriptor.descriptor() );
+        }
+        return recordType().ordinal() - other.recordType().ordinal();
```

Comment:
We checked the docs at https://docs.oracle.com/javase/7/docs/api/java/lang/Comparable.html#compareTo(T) and it is clear that compareTo is allowed to return values other than -1, 0 and +1. They explicitly describe rules that show that non-normalized values can be normalized after the return from compareTo to achieve certain properties. Also the code we used here was pre-existing code in the code-base, so we are not introducing new behaviour.